--- ../../upstream/vyos-build/./scripts/check-qemu-install
+++ ./scripts/check-qemu-install
@@ -868,7 +868,7 @@
     # Verify /etc/os-release via lsb_release
     #################################################
     c.sendline('lsb_release --short --id 2>/dev/null')
-    c.expect('VyOS')
+    c.expect('Tbm-vyos')
     if os.path.isfile(manifest_file):
         c.sendline('lsb_release --short --release 2>/dev/null')
         c.expect(vyos_version)
--- ../../upstream/vyos-build/./scripts/image-build/build-vyos-image
+++ ./scripts/image-build/build-vyos-image
@@ -468,12 +468,12 @@
 
         # Multi line strings needs to be un-indented to not have leading
         # whitespaces in the resulting file
-        os_release = f"""PRETTY_NAME="VyOS {version} ({build_config['release_train']})"
-NAME="VyOS"
+        os_release = f"""PRETTY_NAME="TBM VyOS {version} ({build_config['release_train']})"
+NAME="TBM VyOS"
 VERSION_ID="{version}"
 VERSION="{version} ({build_config['release_train']})"
 VERSION_CODENAME={build_defaults['release_train']}
-ID=vyos
+ID=tbm-vyos
 BUILD_ID="{build_git}"
 HOME_URL="{build_defaults['website_url']}"
 SUPPORT_URL="{build_defaults['support_url']}"
--- ../../upstream/vyos-build/./data/defaults.toml
+++ ./data/defaults.toml
@@ -21,7 +21,7 @@
 squashfs_compression_type = "xz -Xbcj x86 -b 256k -always-use-fragments -no-recovery"
 
 website_url = "https://vyos.io"
-support_url = "https://support.vyos.io"
+support_url = "https://tech.bymatt.au"
 bugtracker_url = "https://vyos.dev"
 documentation_url = "https://docs.vyos.io/en/latest"
 project_news_url = "https://blog.vyos.io"
