--- ../../upstream/vyos-build/./scripts/check-qemu-install	2026-01-16 12:27:13
+++ ./scripts/check-qemu-install	2026-01-16 12:57:54
@@ -43,12 +43,14 @@
 import re
 import shutil
 import json
+import asyncio
 
 from io import BytesIO
 from datetime import datetime
 
 import tomli
 import pexpect
+from qemu.qmp import QMPClient
 
 EXCEPTION = 0
 DISK_IMAGE_EXTENSION = '.raw'
@@ -243,7 +245,7 @@
         {vga} {vnc}\
         -uuid {uuid} \
         {enable_kvm} \
-        -monitor unix:/tmp/qemu-monitor-socket-{disk_img},server,nowait \
+        -qmp unix:/tmp/qemu-monitor-socket-{disk_img},server,nowait \
         -netdev user,id=n0,net=192.0.2.0/24,dhcpstart=192.0.2.101,dns=192.0.2.10 -device virtio-net-pci,netdev=n0,mac={macbase}:00,romfile="",host_mtu=1500 \
         -netdev user,id=n1 -device virtio-net-pci,netdev=n1,mac={macbase}:01,romfile="",host_mtu=1500 \
         -netdev user,id=n2 -device virtio-net-pci,netdev=n2,mac={macbase}:02,romfile="",host_mtu=1500 \
@@ -301,6 +303,16 @@
     c.expect(op_mode_prompt)
     log.info('Logged in!')
 
+async def qmp_eject_cdrom(socket_path, device_id, log):
+    log.info(f'QMP: ejecting {device_id}')
+    client = QMPClient('qemu-monitor')
+    try:
+        await client.connect(socket_path)
+        await client.execute('eject', {'device': device_id, 'force': True})
+        await client.disconnect()
+    except Exception as e:
+        log.error(f'QMP: failed to eject {device_id}: {e}')
+
 # Setting up logger
 log = logging.getLogger()
 log.setLevel(logging.DEBUG)
@@ -381,7 +393,7 @@
             '--ctrl', f'type=unixio,path={tpm_folder}/swtpm-sock', '--tpm2', '--log', 'level=1'
         ])
 
-    from multiprocessing import Process
+    from multiprocess import Process
     tpm_process = Process(target=swtpm_thread)
     tpm_process.start()
     return tpm_process
@@ -611,11 +623,11 @@
         basic_cli_tests(c)
 
         # Valida assigned IPv4 and IPv6 addresses
-        c.sendline('ip --json addr show dev eth0 | jq -r \'.[0].addr_info[] | select(.family=="inet") | "\(.local)/\(.prefixlen)"\'')
+        c.sendline('ip --json addr show dev eth0 | jq -r \'.[0].addr_info[] | select(.family=="inet") | "\\(.local)/\\(.prefixlen)"\'')
         c.expect(CI_CONFIG_ARGS['eth0_ipv4'])
         c.expect(op_mode_prompt)
 
-        c.sendline('ip --json addr show dev eth0 | jq -r \'.[0].addr_info[] | select(.family=="inet6" and .scope=="global" and (.temporary|not)) | "\(.local)/\(.prefixlen)"\'')
+        c.sendline('ip --json addr show dev eth0 | jq -r \'.[0].addr_info[] | select(.family=="inet6" and .scope=="global" and (.temporary|not)) | "\\(.local)/\\(.prefixlen)"\'')
         c.expect(CI_CONFIG_ARGS['eth0_ipv6'])
         c.expect(op_mode_prompt)
 
@@ -703,7 +715,18 @@
         c.expect('input password again:.*')
         c.sendline(mok_password)
         c.expect(op_mode_prompt)
+
+    #################################################
+    # Removing CD installation media
+    #################################################
+    time.sleep(2)
+    log.info('eject installation media')
+    socket_path = f'/tmp/qemu-monitor-socket-{args.disk}'
+    asyncio.run(qmp_eject_cdrom(socket_path, 'drive-cd1', log))
 
+    #################################################
+    # Rebooting after installation
+    #################################################
     log.info('system installed, rebooting')
     c.sendline('reboot now')
 
@@ -713,7 +736,7 @@
     if args.sbtest:
         log.info('Install Secure Boot Machine Owner Key')
         MOK_SLEEP = 0.5
-        c.expect('BdsDxe: starting Boot00.*')
+        c.expect('.*Press any key to perform MOK management.*')
         time.sleep(3)
         # press any key
         c.send(KEY_RETURN)
@@ -751,13 +774,6 @@
     if args.sbtest:
         log.info('Enable UEFI Secure Boot for initial installation')
         toggleUEFISecureBoot(c)
-
-    #################################################
-    # Removing CD installation media
-    #################################################
-    time.sleep(2)
-    log.info('eject installation media')
-    os.system(f'echo "eject -f drive-cd1" | socat - unix-connect:/tmp/qemu-monitor-socket-{args.disk}')
 
     #################################################
     # Booting installed system
@@ -859,7 +875,7 @@
 
         log.info('Encrypting config to TPM')
         c.sendline('encryption enable')
-        c.expect('Automatically generate a recovery key\?.*')
+        c.expect(r'Automatically generate a recovery key\?.*')
         c.sendline('n')
         c.expect('Enter recovery key: ')
         c.sendline(test_recovery_key)
@@ -988,7 +1004,7 @@
 
         log.info('Encrypting config')
         c.sendline('encryption enable')
-        c.expect('Are you sure you want to proceed\?.*')
+        c.expect(r'Are you sure you want to proceed\?.*')
         c.sendline('y')
         c.expect('Enter key: ')
         c.sendline(test_recovery_key)
@@ -1121,7 +1137,7 @@
         if args.match:
             # Remove tests that we don't want to run
             match_str = '-o '.join([f'-name "test_*{name}*.py" ' for name in args.match.split("|")]).strip()
-            c.sendline(f'sudo find {smoketests_path} -maxdepth 2 -type f -name test_* ! \( {match_str} \) -delete')
+            c.sendline(fr'sudo find {smoketests_path} -maxdepth 2 -type f -name test_* ! \( {match_str} \) -delete')
             c.expect(op_mode_prompt)
         if args.no_interfaces:
             # remove interface tests as they consume a lot of time
@@ -1152,7 +1168,7 @@
             raise Exception(tmp)
 
         c.sendline('echo EXITCODE:$\x16?')
-        i = c.expect(['EXITCODE:0', 'EXITCODE:\d+'])
+        i = c.expect(['EXITCODE:0', r'EXITCODE:\d+'])
         if i == 0:
             log.info('Smoketest finished successfully!')
             pass
@@ -1213,7 +1229,7 @@
             raise Exception(tmp)
 
         c.sendline('echo EXITCODE:$\x16?')
-        i = c.expect(['EXITCODE:0', 'EXITCODE:\d+'])
+        i = c.expect(['EXITCODE:0', r'EXITCODE:\d+'])
         if i == 0:
             log.info('Configtest finished successfully!')
             pass
--- ../../upstream/vyos-build/./scripts/image-build/build-vyos-image	2026-01-07 11:08:43
+++ ./scripts/image-build/build-vyos-image	2026-01-16 12:57:54
@@ -341,7 +341,7 @@
 
     ## Convert the image_format field to a single-item list if it's a scalar
     ## (like `image_format = "iso"`)
-    if type(build_config["image_format"]) != list:
+    if not isinstance(build_config["image_format"], list):
         build_config["image_format"] = [ build_config["image_format"] ]
 
     ## If the user didn't explicitly specify what extensions build artifact should have,
@@ -352,7 +352,7 @@
         build_config["artifact_format"] = build_config["image_format"]
     else:
         # If the option is there, also make it list if it's a scalar
-        if type(build_config["artifact_format"]) != list:
+        if not isinstance(build_config["artifact_format"], list):
             build_config["artifact_format"] = [ build_config["artifact_format"] ]
 
     ## Dump the complete config if the user enabled debug mode
@@ -462,12 +462,12 @@
 
         # Multi line strings needs to be un-indented to not have leading
         # whitespaces in the resulting file
-        os_release = f"""PRETTY_NAME="VyOS {version} ({build_config['release_train']})"
-NAME="VyOS"
+        os_release = f"""PRETTY_NAME="TBM VyOS {version} ({build_config['release_train']})"
+NAME="TBM VyOS"
 VERSION_ID="{version}"
 VERSION="{version} ({build_config['release_train']})"
 VERSION_CODENAME={build_defaults['release_train']}
-ID=vyos
+ID=tbm-vyos
 BUILD_ID="{build_git}"
 HOME_URL="{build_defaults['website_url']}"
 SUPPORT_URL="{build_defaults['support_url']}"
@@ -738,7 +738,8 @@
 
     # os.path.splitext returns extensions with dots,
     # so we need to remove the dots, hence [1:]
-    is_artifact = lambda f: os.path.splitext(f)[-1][1:] in build_config['artifact_format']
+    def is_artifact(f):
+        return os.path.splitext(f)[-1][1:] in build_config['artifact_format']
 
     manifest['artifacts'] = list(filter(is_artifact, manifest['artifacts']))
 
--- ../../upstream/vyos-build/./data/defaults.toml	2026-01-07 11:08:43
+++ ./data/defaults.toml	2026-01-16 12:57:54
@@ -21,7 +21,7 @@
 squashfs_compression_type = "xz -Xbcj x86 -b 256k -always-use-fragments -no-recovery"
 
 website_url = "https://vyos.io"
-support_url = "https://support.vyos.io"
+support_url = "https://tech.bymatt.au"
 bugtracker_url = "https://vyos.dev"
 documentation_url = "https://docs.vyos.io/en/latest"
 project_news_url = "https://blog.vyos.io"
