--- ../../upstream/vyos-1x/./smoketest/scripts/cli/test_op-mode_show.py	2026-01-07 11:11:25
+++ ./smoketest/scripts/cli/test_op-mode_show.py	2026-01-07 11:08:40
@@ -29,7 +29,7 @@
         tmp = self.op_mode(base_path + ['version'])
         # Validate
         version = get_version()
-        self.assertIn(f'Version:          VyOS {version}', tmp)
+        self.assertIn(f'Version:          TBM VyOS {version}', tmp)
 
     def test_op_mode_show_version_kernel(self):
         # Retrieve output of "show version" OP-mode command
--- ../../upstream/vyos-1x/./data/templates/login/default_motd.j2	2026-01-07 11:11:24
+++ ./data/templates/login/default_motd.j2	2026-01-07 18:55:19
@@ -1,8 +1,8 @@
-Welcome to VyOS!
+Welcome to TBM VyOS!
 
    ┌── ┐
-   . VyOS {{ version_data.version }}
-   └ ──┘  {{ version_data.release_train }}
+   . TBM VyOS {{ version_data.version }}
+   └ ──┘
 
 {% if version_data.lts_build %}
  * Support portal: {{ version_data.support_url }}
@@ -13,5 +13,5 @@
 
 You can change this banner using "set system login banner post-login" command.
 
-VyOS is a free software distribution that includes multiple components,
+TBM VyOS is a free software distribution that includes multiple components,
 you can check individual component licenses under /usr/share/doc/*/copyright
--- ../../upstream/vyos-1x/./src/op_mode/version.py	2026-01-07 11:11:25
+++ ./src/op_mode/version.py	2026-01-07 11:08:40
@@ -31,7 +31,7 @@
 from jinja2 import Template
 
 version_output_tmpl = """
-Version:          VyOS {{version}}
+Version:          TBM VyOS {{version}}
 Release train:    {{release_train}}
 Release flavor:   {{flavor}}
 
@@ -54,6 +54,7 @@
 Hardware UUID:    {{hardware_uuid}}
 
 Copyright:        VyOS maintainers and contributors
+                  Tech By Matt <info@tech.bymatt.au>
 {%- if limerick %}
 {{limerick}}
 {% endif -%}
@@ -94,4 +95,3 @@
     except (ValueError, vyos.opmode.Error) as e:
         print(e)
         sys.exit(1)
-
--- ../../upstream/vyos-1x/./src/op_mode/image_installer.py	2026-01-07 11:11:25
+++ ./src/op_mode/image_installer.py	2026-01-07 18:59:21
@@ -79,8 +79,8 @@
 MSG_ERR_MISSING_FLAVOR: str = 'The new image version data does not specify flavor, cannot check compatibility (is it a legacy release image?)'
 MSG_ERR_CORRUPT_CURRENT_IMAGE: str = 'Version data in the current image is malformed: missing flavor and/or architecture fields. Upgrade compatibility cannot be checked.'
 MSG_ERR_UNSUPPORTED_SIGNATURE_TYPE: str = 'Unsupported signature type, signature cannot be verified.'
-MSG_INFO_INSTALL_WELCOME: str = 'Welcome to VyOS installation!\nThis command will install VyOS to your permanent storage.'
-MSG_INFO_INSTALL_EXIT: str = 'Exiting from VyOS installation'
+MSG_INFO_INSTALL_WELCOME: str = 'Welcome to TBM VyOS installation!\nThis command will install TBM VyOS to your permanent storage.'
+MSG_INFO_INSTALL_EXIT: str = 'Exiting from TBM VyOS installation'
 MSG_INFO_INSTALL_SUCCESS: str = 'The image installed successfully; please reboot now.'
 MSG_INFO_INSTALL_DISKS_LIST: str = 'The following disks were found:'
 MSG_INFO_INSTALL_DISK_SELECT: str = 'Which one should be used for installation?'
--- ../../upstream/vyos-1x/./src/conf_mode/system_login_banner.py	2026-01-07 11:11:25
+++ ./src/conf_mode/system_login_banner.py	2026-01-07 11:08:40
@@ -31,7 +31,7 @@
 POSTLOGIN_VYOS_FILE = r'/run/motd.d/01-vyos-nonproduction'
 
 default_config_data = {
-    'issue': 'Welcome to VyOS - \\n \\l\n\n',
+    'issue': 'Welcome to TBM VyOS - \\n \\l\n\n',
     'issue_net': ''
 }
 
