--- ../../upstream/vyos-1x/./smoketest/scripts/cli/test_op-mode_show.py	2026-01-07 11:11:25
+++ ./smoketest/scripts/cli/test_op-mode_show.py	2026-01-07 11:08:40
@@ -29,7 +29,7 @@
         tmp = self.op_mode(base_path + ['version'])
         # Validate
         version = get_version()
-        self.assertIn(f'Version:          VyOS {version}', tmp)
+        self.assertIn(f'Version:          TBM VyOS {version}', tmp)
 
     def test_op_mode_show_version_kernel(self):
         # Retrieve output of "show version" OP-mode command
--- ../../upstream/vyos-1x/./data/templates/login/default_motd.j2	2026-01-07 11:11:24
+++ ./data/templates/login/default_motd.j2	2026-01-07 11:08:40
@@ -1,8 +1,8 @@
-Welcome to VyOS!
+Welcome to TBM VyOS!
 
    ┌── ┐
-   . VyOS {{ version_data.version }}
-   └ ──┘  {{ version_data.release_train }}
+   . TBM VyOS {{ version_data.version }}
+   └ ──┘      {{ version_data.release_train }}
 
 {% if version_data.lts_build %}
  * Support portal: {{ version_data.support_url }}
@@ -13,5 +13,5 @@
 
 You can change this banner using "set system login banner post-login" command.
 
-VyOS is a free software distribution that includes multiple components,
+TBM VyOS is a free software distribution that includes multiple components,
 you can check individual component licenses under /usr/share/doc/*/copyright
--- ../../upstream/vyos-1x/./src/op_mode/version.py	2026-01-07 11:11:25
+++ ./src/op_mode/version.py	2026-01-07 11:08:40
@@ -31,7 +31,7 @@
 from jinja2 import Template
 
 version_output_tmpl = """
-Version:          VyOS {{version}}
+Version:          TBM VyOS {{version}}
 Release train:    {{release_train}}
 Release flavor:   {{flavor}}
 
@@ -54,6 +54,7 @@
 Hardware UUID:    {{hardware_uuid}}
 
 Copyright:        VyOS maintainers and contributors
+                  Tech By Matt <info@tech.bymatt.au>
 {%- if limerick %}
 {{limerick}}
 {% endif -%}
@@ -94,4 +95,3 @@
     except (ValueError, vyos.opmode.Error) as e:
         print(e)
         sys.exit(1)
-
--- ../../upstream/vyos-1x/./src/op_mode/image_installer.py	2026-01-07 11:11:25
+++ ./src/op_mode/image_installer.py	2026-01-07 11:08:40
@@ -63,13 +63,11 @@
 from vyos.utils.auth import get_local_users
 from vyos.utils.auth import get_user_home_dir
 from vyos.version import get_version_data
-from vyos.config_mgmt import unsaved_commits
 
 # define text messages
 MSG_ERR_NOT_LIVE: str = 'The system is already installed. Please use "add system image" instead.'
 MSG_ERR_LIVE: str = 'The system is in live-boot mode. Please use "install image" instead.'
 MSG_ERR_NOT_ENOUGH_SPACE: str = 'Image upgrade requires at least 2GB of free drive space.'
-MSG_ERR_UNSAVED_COMMITS: str = 'There are unsaved changes to the configuration. Either save or revert before upgrade.'
 MSG_ERR_NO_DISK: str = 'No suitable disk was found. There must be at least one disk of 2GB or greater size.'
 MSG_ERR_IMPROPER_IMAGE: str = 'Missing sha256sum.txt.\nEither this image is corrupted, or of era 1.2.x (md5sum) and would downgrade image tools;\ndisallowed in either case.'
 MSG_ERR_INCOMPATIBLE_IMAGE: str = 'Image compatibility check failed, aborting installation.'
@@ -79,8 +77,8 @@
 MSG_ERR_MISSING_FLAVOR: str = 'The new image version data does not specify flavor, cannot check compatibility (is it a legacy release image?)'
 MSG_ERR_CORRUPT_CURRENT_IMAGE: str = 'Version data in the current image is malformed: missing flavor and/or architecture fields. Upgrade compatibility cannot be checked.'
 MSG_ERR_UNSUPPORTED_SIGNATURE_TYPE: str = 'Unsupported signature type, signature cannot be verified.'
-MSG_INFO_INSTALL_WELCOME: str = 'Welcome to VyOS installation!\nThis command will install VyOS to your permanent storage.'
-MSG_INFO_INSTALL_EXIT: str = 'Exiting from VyOS installation'
+MSG_INFO_INSTALL_WELCOME: str = 'Welcome to TBM VyOS installation!\nThis command will install TBM VyOS to your permanent storage.'
+MSG_INFO_INSTALL_EXIT: str = 'Exiting from TBM VyOS installation'
 MSG_INFO_INSTALL_SUCCESS: str = 'The image installed successfully; please reboot now.'
 MSG_INFO_INSTALL_DISKS_LIST: str = 'The following disks were found:'
 MSG_INFO_INSTALL_DISK_SELECT: str = 'Which one should be used for installation?'
@@ -93,7 +91,6 @@
 MSG_INPUT_CONFIG_FOUND: str = 'An active configuration was found. Would you like to copy it to the new image?'
 MSG_INPUT_CONFIG_CHOICE: str = 'The following config files are available for boot:'
 MSG_INPUT_CONFIG_CHOOSE: str = 'Which file would you like as boot config?'
-MSG_INPUT_UNSAVED_COMMITS: str = 'There are unsaved changes to the configuration. They will not be copied to the new image. Continue without saving?'
 MSG_INPUT_IMAGE_NAME: str = 'What would you like to name this image?'
 MSG_INPUT_IMAGE_NAME_TAKEN: str = 'There is already an installed image by that name; please choose again'
 MSG_INPUT_IMAGE_DEFAULT: str = 'Would you like to set the new image as the default one for boot?'
@@ -277,7 +274,7 @@
             if disk.partition_mount(partition, mnt_tmp):
                 if Path(mnt_tmp + '/boot').exists():
                     for path in Path(mnt_tmp + '/boot').iterdir():
-                        if path.joinpath('rw/opt/vyatta/etc/config/.vyatta_config').exists():
+                        if path.joinpath('rw/config/.vyatta_config').exists():
                             image_data.append((path.name, partition))
                 if Path(mnt_tmp + '/luks').exists():
                     for path in Path(mnt_tmp + '/luks').iterdir():
@@ -330,7 +327,7 @@
     disk.partition_mount(image_drive, mnt_tmp)
 
     if not encrypted:
-        copytree(f'{mnt_tmp}/boot/{image_name}/rw/opt/vyatta/etc/config', mnt_config)
+        copytree(f'{mnt_tmp}/boot/{image_name}/rw/config', mnt_config)
     else:
         copy(f'{mnt_tmp}/luks/{image_name}', mnt_encrypted_config)
 
@@ -524,10 +521,6 @@
     if 'quiet' in kernel_options:
         cmdline_options.append('quiet')
 
-    # Early reboot on kernel panic via kernel cmdline (must match system_option.py)
-    if dict_search('system.option.reboot-on-panic', config_dict) is not None:
-        cmdline_options.append('panic=60')
-
     if 'disable-hpet' in kernel_options:
         cmdline_options.append('hpet=disable')
 
@@ -1105,13 +1098,6 @@
     if disk_usage('/').free < (2 * 1024**3):
         exit(MSG_ERR_NOT_ENOUGH_SPACE)
 
-    if unsaved_commits():
-        if not no_prompt:
-            if not ask_yes_no(MSG_INPUT_UNSAVED_COMMITS, default=False):
-                exit()
-        else:
-            exit(MSG_ERR_UNSAVED_COMMITS)
-
     environ['REMOTE_USERNAME'] = username
     environ['REMOTE_PASSWORD'] = password
 
--- ../../upstream/vyos-1x/./src/conf_mode/system_login_banner.py	2026-01-07 11:11:25
+++ ./src/conf_mode/system_login_banner.py	2026-01-07 11:08:40
@@ -31,7 +31,7 @@
 POSTLOGIN_VYOS_FILE = r'/run/motd.d/01-vyos-nonproduction'
 
 default_config_data = {
-    'issue': 'Welcome to VyOS - \\n \\l\n\n',
+    'issue': 'Welcome to TBM VyOS - \\n \\l\n\n',
     'issue_net': ''
 }
 
